### PR TITLE
Add filtering and sorting for list operations of Bitbucket connector

### PR DESCRIPTION
**Pull request by SanduDS**
## Purpose
Adding filtering and sorting facility for the listing operations such as list issues, workspaces, pull requests etc.
Fixes https://github.com/wso2-enterprise/choreo/issues/6161

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: openapi/bitbucket/client.bal [hunk 1]
```diff
@@ -37,6 +39,41 @@ public client class Client {
         http:Client httpEp = check new (serviceUrl, { auth: clientConfig.authConfig, secureSocket: secureSocketConfig });
         self.clientEp = httpEp;
     }
+    # Returns a list of all repositories
+    #
+    # + workspace - This can either be the workspace ID (slug) or the workspace UUID
+    # + role - Filters the result based on the authenticated user's role on each repository. member-returns repositories to which the user has explicit read access. contributor- returns repositories to which the user has explicit write access. admin- returns repositories to which the user has explicit administrator access. owner-returns all repositories owned by the current user
```
Comment (LakshanSS) [inline openapi/bitbucket/client.bal:45]: Need to break this after 120 characters.
Check other places also

File: openapi/bitbucket/client.bal [hunk 2]
```diff
@@ -331,3 +374,37 @@ public client class Client {
...
+            _ = queryParam.remove(key);
+        } else {
+            if  string:startsWith( key, "'") {
+                 param[param.length()] = string:substring(key, 1, key.length());
+            } else {
+                param[param.length()] = key;
+            }
+            param[param.length()] = "=";
+            if  value  is  string {
+                string updateV =  checkpanic url:encode(value, "UTF-8");
```
Comment (LakshanSS) [inline openapi/bitbucket/client.bal:396]: Is there a reason to use checkpanic instead of check?

File: openapi/bitbucket/client.bal [hunk 3]
```diff
@@ -331,3 +374,37 @@ public client class Client {
          _ = check self.clientEp-> delete(path, request, targetType =http:Response);
     }
 }
+
+# Generate query path with query parameter.
+#
+# + queryParam - Query parameter map
+# + return - Returns generated Path or error at failure of client initialization
+isolated function  getPathForQueryParam(map<anydata>   queryParam)  returns  string {
```
Comment (lnash94) [inline openapi/bitbucket/client.bal:382]:
```suggestion
isolated function  getPathForQueryParam(map<anydata>   queryParam)  returns  string|error {
```

File: openapi/bitbucket/client.bal (hunk 2 above)
Comment (lnash94) [inline openapi/bitbucket/client.bal:396]:
```suggestion
                string updateV =  check url:encode(value, "UTF-8");
```

File: openapi/bitbucket/client.bal [hunk 4]
```diff
@@ -147,12 +184,18 @@ public client class Client {
...
     @display {label: "List Issues"}
     remote isolated function listIssues(@display {label: "Repository ID (slug) or UUID"} string repo_slug, 
-                                        @display {label: "Workspace ID (slug) or UUID"} string workspace) returns 
+                                        @display {label: "Workspace ID (slug) or UUID"} string workspace,
+                                        @display {label: "Filtering Query"} string? query = (), 
+                                        @display {label: "Sorting Query"} string? sort = ()) returns 
                                         PaginatedIssues|error {
         string  path = string `/repositories/${workspace}/${repo_slug}/issues`;
+        map<anydata> queryParam = {"q": query, "sort": sort};
+        path = path + getPathForQueryParam(queryParam);
```
Comment (lnash94) [inline openapi/bitbucket/client.bal:198]:
```suggestion
        path = path + check getPathForQueryParam(queryParam);
```

File: openapi/bitbucket/client.bal [hunk 5]
```diff
@@ -37,6 +39,41 @@ public client class Client {
...
+    # + sort - Field by which the results should be sorted as https://developer.atlassian.com/bitbucket/api/2/reference/meta/filtering
+    # + return - The repositories owned by the specified account.
+    @display {label: "List Workspaces"}
+    remote isolated function listWorkspaces(@display {label: "Workspace ID (slug) or UUID"} string workspace, 
+                                            @display {label: "User Role"} string? role = (), 
+                                            @display {label: "Filtering Query"} string? query = (), 
+                                            @display {label: "Sorting Query"} string? sort = ()) returns PaginatedRepositories|error {
+        string  path = string `/repositories/${workspace}`;
+        map<anydata> queryParam = {"role": role, "q": query, "sort": sort};
+        path = path + getPathForQueryParam(queryParam);
```
Comment (lnash94) [inline openapi/bitbucket/client.bal:56]:
```suggestion
        path = path + check getPathForQueryParam(queryParam);
```

Comment (lnash94) [reply openapi/bitbucket/client.bal:56]: Shall we check other places using `getPathForQueryParam(queryParam)` and change it with `check` ? :)

File: openapi/bitbucket/client.bal (hunk 1 above)
Comment (indikasampath2000) [reply openapi/bitbucket/client.bal:45]: @SanduDS It seems you are not using the latest tool to generate the code. We have patched it to fix these issues.

File: openapi/bitbucket/client.bal (hunk 5 above)
Comment (indikasampath2000) [reply openapi/bitbucket/client.bal:56]: @lnash94 do we need to fix it manually? Haven't we patched the alpha5 tool?

File: openapi/bitbucket/client.bal (hunk 2 above)
Comment (SanduDS) [reply openapi/bitbucket/client.bal:396]: This is an auto-generated code, I'll try with the latest tool again

Comment (SanduDS) [reply openapi/bitbucket/client.bal]: Resolved manually

**Review by indikasampath2000 — APPROVED** [review]

**Pull request merged**